components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthAdminService.java

```diff
@@ -619,18 +622,53 @@ public void removeOAuthApplicationData(String consumerKey) throws IdentityOAuthA
     }
 
     /**
-     * Revoke authorization for OAuth apps by resource owners
+     * Get apps that are authorized by the given user
+     *
+     * @return OAuth applications authorized by the user that have tokens in ACTIVE or EXPIRED state
+     */
+    public OAuthConsumerAppDTO[] getAppsAuthorizedByUser() throws IdentityOAuthAdminException {
+
+        return getoAuthConsumerAppDTOS(PrivilegedCarbonContext.getThreadLocalCarbonContext().getUsername());
+    }
+
+    public OAuthRevocationResponseDTO[] revokeAuthorization(String username) throws IdentityOAuthAdminException {
+
+        OAuthConsumerAppDTO oAuthConsumerAppDTOs[] = getAuthorizedApps(username);
+        OAuthRevocationResponseDTO oAuthRevocationResponseDTO[] = new
+                OAuthRevocationResponseDTO[oAuthConsumerAppDTOs.length];
```

Comment:
Shall we define oAuthRevocationResponseDTO as a list and when return converts it to array.